packages/flutter_tools/lib/src/build_system/targets/macos.dart

```diff
@@ -128,6 +125,48 @@ class UnpackMacOS extends Target {
   }
 }
 
+/// Unpack the release prebuilt engine framework.
+class ReleaseUnpackMacOS extends UnpackMacOS {
+  const ReleaseUnpackMacOS();
+
+  @override
+  String get name => 'release_unpack_macos';
+
+  @override
+  List<Source> get inputs => const <Source>[
```

Comment:
It doesn't seem like this construction (where the inputs are different, but the destination is the same) actually works; am I missing something? The Debug and Release sources are unchanging (other than on Flutter updates), which means that they are ~always going to be older than the last unpacked version. Why would these steps run after changing build modes?

Reply:
Age doesn't matter here because I use hashed file contents. These hashes are stored per build configuration, so when running a different config the tool will notice that the output hashes have since changed and the build needs to be rerun.